server.js

```diff
@@ -96,20 +95,21 @@ function searchWeather(request, response) {
   const lat = request.query.data.latitude;
   const lng = request.query.data.longitude;
   const locationName = request.query.data;
-  const qryString = `SELECT * FROM weathers WHERE location_id=${locationName.id}`;
-  
   const url = `https://api.darksky.net/forecast/${WEATHER_API_KEY}/${lat},${lng}`;
 
-  checkForExistance(qryString, ifExistW, noExistW, locationName, url, response);
+  client.query(`SELECT id FROM locations WHERE search_query='${locationName.search_query}'`).then(sqlRes => {
+    const qryString = `SELECT * FROM weathers WHERE location_id=${sqlRes.rows[0].id}`;
+    checkForExistance(qryString, ifExistW, noExistW, sqlRes.rows[0].id, url, response);
+  });
+
 }
 
 // check DB for existance
-function checkForExistance(qryString, doesExist, noExist, locationName, url, response) {
+function checkForExistance(qryString, doesExist, noExist, locationId, url, response) {
```

Comment:
i like you you built this into seperate small functions that eacd do a small part, it looks very clean and would make maintaining it in the future easy